selfservice/strategy/link/sender.go

```diff
@@ -137,13 +138,20 @@ func (s *Sender) SendVerificationTokenTo(ctx context.Context, f *verification.Fl
 		WithSensitiveField("verification_link_token", token.Token).
 		Info("Sending out verification email with verification link.")
 
-	return s.send(ctx, string(address.Via), templates.NewVerificationValid(s.r.Config(ctx),
+	if err := s.send(ctx, string(address.Via), templates.NewVerificationValid(s.r.Config(ctx),
 		&templates.VerificationValidModel{To: address.Value, VerificationURL: urlx.CopyWithQuery(
 			urlx.AppendPaths(s.r.Config(ctx).SelfPublicURL(nil), verification.RouteSubmitFlow),
 			url.Values{
 				"flow":  {f.ID.String()},
 				"token": {token.Token},
-			}).String()}))
+			}).String()})); err != nil {
+		return err
+	}
+	address.Status = identity.VerifiableAddressStatusPending
```

Comment:
Awesome! Could you now please add a test to verify that this is working as intended? Basically sending an email, checking if the status was updated, and then calling the hook again - and making sure the email is not sent twice!

Reply:
> Awesome! Could you now please add a test to verify that this is working as intended? Basically sending an email, checking if the status was updated, and then calling the hook again - and making sure the email is not sent twice!

added